server/src/main/java/org/apache/iotdb/db/writelog/manager/MultiFileLogNodeManager.java

```diff
@@ -90,9 +96,39 @@ public WriteLogNode getNode(String identifier, Supplier<ByteBuffer[]> supplier)
       node = new ExclusiveWriteLogNode(identifier);
       WriteLogNode oldNode = nodeMap.putIfAbsent(identifier, node);
       if (oldNode != null) {
-        return oldNode;
+        node = oldNode;
       } else {
-        node.initBuffer(supplier.get());
+        ByteBuffer[] buffers = null;
+        int sleepTimeInMs = 0;
+        // try to get bytebuffer repeatedly
+        while (buffers == null) {
+          try {
+            buffers = supplier.get();
+          } catch (OutOfMemoryError e) {
+            // log error if this is the first time
+            if (sleepTimeInMs == 0) {
+              logger.error(
+                  "OOM occurs when allocate bytebuffer for wal, please reduce wal_buffer_size or storage groups number",
+                  e);
+            }
+            // sleep awhile and then try again
+            try {
+              Thread.sleep(REGISTER_BUFFER_SLEEP_INTERVAL);
+              sleepTimeInMs += REGISTER_BUFFER_SLEEP_INTERVAL;
+            } catch (InterruptedException interruptedException) {
+              e.addSuppressed(interruptedException);
+              nodeMap.remove(identifier);
+              throw e;
+            }
+            // sleep too long, throw exception
+            if (sleepTimeInMs >= REGISTER_BUFFER_REJECT_THRESHOLD) {
+              nodeMap.remove(identifier);
+              throw e;
+            }
```

Comment:
just judge whether the buffers is null, and no need to catch OOM here.

Reply:
fixed